src/request/plan.rs

```diff
@@ -249,6 +249,35 @@ where
     }
 }
 
+pub struct PropagateError<P: Plan> {
```

Comment:
Could you document the plan please? I also think the name is not quite right, maybe `ExtractError` or something is better?

Reply:
My first thought was `ExtractError` too, but there exists a similar name `extract_errors`.

Reply:
It's in a pretty different part of the system, so I think it should be ok